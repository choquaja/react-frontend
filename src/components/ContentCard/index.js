import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { themeColor } from '../../services/theme';

const isColumn = props => props.column && css`
  margin: 0 -1rem;
`;

const isTight = props => props.tight && css`
  padding: 0;
`;

const ContentCard = styled.div`
  padding: 1rem 1rem 2rem;
  ${isTight}
  border-radius: 3px;
  background-color: ${themeColor('white')};
  box-shadow: rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.117647) 0px 1px 4px;
  overflow: hidden;
  position: relative;
  ${isColumn}
`;

ContentCard.propTypes = {
  column: PropTypes.bool,
  tight: PropTypes.bool,
};

export default ContentCard;
