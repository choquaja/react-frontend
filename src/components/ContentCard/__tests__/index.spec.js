import React from 'react';
import { shallow } from 'enzyme';
import ContentCard from '../';

describe('ContentCard', () => {
  it('renders without crashing', () => {
    shallow(<ContentCard />);
  });

  it('renders with the column prop', () => {
    const card = shallow(<ContentCard column />);
    expect(card).toMatchSnapshot();
  });

  it('renders with the tight prop', () => {
    const card = shallow(<ContentCard tight />);
    expect(card).toMatchSnapshot();
  });
});
