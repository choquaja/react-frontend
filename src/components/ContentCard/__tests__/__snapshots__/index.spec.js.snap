// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard renders with the column prop 1`] = `
<div
  className="ContentCard__ContentCard-kKBMhH kUIZGp"
/>
`;

exports[`ContentCard renders with the tight prop 1`] = `
<div
  className="ContentCard__ContentCard-kKBMhH haWMJS"
/>
`;
